tools/llvm-spirv/llvm-spirv.cpp

```diff
@@ -741,8 +741,14 @@ int main(int Ac, char **Av) {
     } else {
       Opts.setDebugInfoEIS(DebugEIS);
       if (DebugEIS.getValue() ==
-          SPIRV::DebugInfoEIS::NonSemantic_Kernel_DebugInfo_100)
+          SPIRV::DebugInfoEIS::NonSemantic_Shader_DebugInfo_200)
         Opts.setAllowExtraDIExpressionsEnabled(true);
+      if (DebugEIS.getValue() ==
+          SPIRV::DebugInfoEIS::NonSemantic_Shader_DebugInfo_200 ||
+          DebugEIS.getValue() ==
+          SPIRV::DebugInfoEIS::NonSemantic_Shader_DebugInfo_200)
+        Opts.setAllowedToUseExtension(
+            SPIRV::ExtensionID::SPV_KHR_non_semantic_info);
```

Comment:
Here we actually implicitly set --spirv-ext=+SPV_KHR_non_semantic_info. Without it addExtension(SPV_KHR_non_semantic_info) would fail.

We could remove it and add to the tests +SPV_KHR_non_semantic_info explicitly, but I'm not sure if it's expected by the users.

Reply:
I'm not sure too. For me explicit extension setting is more logical, but maybe for users it'd be less convenient. 
I don't have preference here, we can proceed with the implemented solution

Reply:
I don't see a reason why we shouldn't enable the extension implicitly while it's guarantied that the instruction set is NonSemantic by its name.